apps/meteor/app/livechat/server/lib/rooms.ts

```diff
@@ -92,6 +101,43 @@ export async function createRoom({
 		}
 	}
 
+	const now = new Date();
+	const roomProps: Omit<IOmnichannelRoom, '_id' | '_updatedAt' | 'priorityWeight' | 'estimatedWaitingTimeQueue' | 'waitingResponse'> = {
+		v: {
+			_id: visitor._id,
+			username: visitor.username,
+			status: visitor.status,
+			name: visitor.name,
+			token: visitor.token,
+			activity: visitor.activity,
+			lastMessageTs: visitor.lastChat?.ts,
+			phone: visitor.phone?.[0]?.phoneNumber ?? undefined,
+		},
+		departmentId: visitor.department,
+		source: roomInfo.source,
+		t: RoomType.LIVE_CHAT,
+		usersCount: 0,
+		msgs: 0,
+		u: {
+			_id: visitor._id,
+			username: visitor.username,
+			name: visitor.name,
+		},
+		ts: now,
+		livechatData: visitor.livechatData,
+		customFields: extraData?.customFields,
+	};
+
+	try {
+		await Apps.self?.triggerEvent(AppEvents.IPreLivechatRoomCreatePrevent, roomProps);
```

Comment:
apparently there isn't an standard way of doing this, but since we allow the event just returning a `boolean` (without throwing an exception) we should look for that scenario as well:
```suggestion
		const prevent = await Apps.self?.triggerEvent(AppEvents.IPreLivechatRoomCreatePrevent, roomProps);
		if (prevent) {
			throw new Meteor.Error('error-app-prevented', 'A Rocket.Chat App prevented the room creation.');
		}
```